### ISSUE TITLE
check_queue.py

### ISSUE THREAD

**Issue by igorxo** [wazuh/wazuh-splunk#911]
| Wazuh | Splunk | Rev |
| ----- | ------- | --- |
| 3.11.4   | 8.0.2.1 | f002026bad55 |

**Description**
After instaling Wazuh app on Splunk I am getting following error in logs. I don't know how this affects app operations.

03-17-2020 08:09:51.952 +0100 ERROR ExecProcessor - message from "/opt/splunk/bin/python2.7 /opt/splunk/etc/apps/SplunkAppForWazuh/bin/check_queue.py" Traceback (most recent call last):
03-17-2020 08:09:51.952 +0100 ERROR ExecProcessor - message from "/opt/splunk/bin/python2.7 /opt/splunk/etc/apps/SplunkAppForWazuh/bin/check_queue.py"   File "/opt/splunk/etc/apps/SplunkAppForWazuh/bin/check_queue.py", line 18, in <module>
03-17-2020 08:09:51.953 +0100 ERROR ExecProcessor - message from "/opt/splunk/bin/python2.7 /opt/splunk/etc/apps/SplunkAppForWazuh/bin/check_queue.py"     import jsonbak
03-17-2020 08:09:51.953 +0100 ERROR ExecProcessor - message from "/opt/splunk/bin/python2.7 /opt/splunk/etc/apps/SplunkAppForWazuh/bin/check_queue.py"   File "/opt/splunk/etc/apps/SplunkAppForWazuh/bin/jsonbak/__init__.py", line 120
03-17-2020 08:09:51.953 +0100 ERROR ExecProcessor - message from "/opt/splunk/bin/python2.7 /opt/splunk/etc/apps/SplunkAppForWazuh/bin/check_queue.py"     def dump(obj, fp, *, skipkeys=False, ensure_ascii=True, check_circular=True,
03-17-2020 08:09:51.953 +0100 ERROR ExecProcessor - message from "/opt/splunk/bin/python2.7 /opt/splunk/etc/apps/SplunkAppForWazuh/bin/check_queue.py"                        ^
03-17-2020 08:09:51.953 +0100 ERROR ExecProcessor - message from "/opt/splunk/bin/python2.7 /opt/splunk/etc/apps/SplunkAppForWazuh/bin/check_queue.py" SyntaxError: invalid syntax
 

**Steps to reproduce**

1. Install Wazuh app on Splunk
2. Check splunkd.log

**Comment by pablotr9** [comment]
Hi @igorxo ,

Thanks for reporting that error, I have been checking the latest package and indeed there's an error in the check_queue function.
The function of this method is to create a queue of jobs in the KeyValue store of Splunk so the app can manage multiple requests and wait for the first job to be finished until the second job is sent. This method was only being used when restarting the Wazuh Manager multiple times from the Wazuh Splunk App.
This error isn't expected to cause any problem while using the App, we will fix it as soon as possible. 

Please let us know if you find any other problem, and thanks for reporting the error!

Best Regards,

**Comment by GuardianCosmos** [comment]
@pablotr9 I'm noticing the same issue followed directly by a TcpOutputProc error the pauses the dataflow, this is pretty consistent and I'm not sure if its related.
With my issue, I'm not getting any of the logs imported into splunk, though I am able to see and manage the agents, no data comes back about them into splunk

**Comment by pablotr9** [comment]
Hi @igorxo @GuardianCosmos , 

This error has been fixed here: #928 , It will be added in the next release,

You can manually fix this by editing the `inputs.conf` file located at `opt/splunk/etc/apps/SplunkAppForWazuh/default/inputs.conf`:

Replacing its content with:
```
[splunktcp://9997]
connection_host = ip

[script://$SPLUNK_HOME/etc/apps/SplunkAppForWazuh/bin/get_agents_status.py]
disabled = false
index = wazuh-monitoring-3x
interval = 5
sourcetype = _json
passAuth = splunk-system-user
python.version = python3

[script://$SPLUNK_HOME/etc/apps/SplunkAppForWazuh/bin/check_queue.py]
disabled = false
interval = 15.0
passAuth = splunk-system-user
python.version = python3

```

This way you can specify the python interpreter version (3.7) which is needed to run these scripts.

I will close this issue, please feel free to reopen this issue or open a new one if the problem persists!

Kind Regards,
Pablo Torres
